lib/selector/parser/ast.go

```diff
@@ -40,13 +43,52 @@ func (l MapAsLabels) Get(labelName string) (value string, present bool) {
 type Selector interface {
 	// Evaluate evaluates the selector against the given labels expressed as a concrete map.
 	Evaluate(labels map[string]string) bool
+
 	// EvaluateLabels evaluates the selector against the given labels expressed as an interface.
 	// This allows for labels that are calculated on the fly.
 	EvaluateLabels(labels Labels) bool
+
 	// String returns a string that represents this selector.
 	String() string
+
 	// UniqueID returns the unique ID that represents this selector.
 	UniqueID() string
+
+	// AcceptVisitor allows an external visitor to modify this selector.
+	AcceptVisitor(v Visitor)
+}
+
+type Visitor interface {
+	Visit(n interface{})
+}
+
+// PrefixVisitor implements the Visitor interface to allow prefixing of
+// label names within a selector.
+type PrefixVisitor struct {
+	Prefix string
+}
+
+func (v PrefixVisitor) Visit(n interface{}) {
+	log.Debugf("PrefixVisitor visiting node %#v", n)
+	switch n.(type) {
```

Comment:
Use `switch np := n.(type)`, then you don't need the type assertions in each case.

Reply:
Neat, I didn't know about that.

So, it does a type check _and_ a cast in one operation?